### PR TITLE
Feat/editing

### PR DESCRIPTION
**Pull request by Maxycle**
Merge after #86 

This is ready to merge provided we don't want to create another release before merging.

### CODE REVIEW

**Review by WillKocur — CHANGES_REQUESTED** [review]
Had a quick look through this one while waiting on the sorting branch, but won't merge until feat/sorting is merged first. Looks good to me! A lot of it is built on in the PR being merged into this one anyway, so I ignored those files that are changed -  e.g., the view files as they have already been reviewed in the later versions. 

Added a couple of small comments, but just little things. Cheers!

File: config/initializers/assets.rb
Comment (Maxycle) [reply config/initializers/assets.rb:15]: I should probably have made some comments because I can't remember why this is here...

**Pull request merged**
